### PR TITLE
Added tensor.is_complex(), is_complex and dtype.is_complex py binding, tensor printing, and dixed the scalar type returned for complex float

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33507 Gradcheck for complex
* #33361 torch.tensor can infer complex dtype now
* **#33268 Added tensor.is_complex(), is_complex and dtype.is_complex py binding, tensor printing, and dixed the scalar type returned for complex float**

Differential Revision: [D19907698](https://our.internmc.facebook.com/intern/diff/D19907698)

### CODE REVIEW

**Comment by anjali411** [comment]
discussion: https://github.com/pytorch/pytorch/pull/33175

File: torch/csrc/utils/python_scalars.h [hunk 1]
```diff
@@ -40,7 +40,10 @@ inline PyObject* load_scalar(void* data, at::ScalarType scalarType) {
     case at::kHalf: return PyFloat_FromDouble(at::convert<double, at::Half>(*(at::Half*)data));
     case at::kFloat: return PyFloat_FromDouble(*(float*)data);
     case at::kDouble: return PyFloat_FromDouble(*(double*)data);
-    case at::kComplexFloat: return PyComplex_FromCComplex(*reinterpret_cast<Py_complex *>((std::complex<float>*)data));
+    case at::kComplexFloat: {
+      std::complex<float>* data_ = (std::complex<float>*)data;
+      return PyComplex_FromDoubles(data_->real(), data_->imag());
+    }
```
Comment (ezyang) [inline torch/csrc/utils/python_scalars.h]: nit: this bugfix could have gotten a PR all on its own

**Review by ezyang — APPROVED** [review]

**Review by apaszke — COMMENTED** [review]
Can't we just have complex be another dtype? I'd rather avoid adding more dtype-specific methods to tensors.

**Comment by ezyang** [comment]
I don't really have a preference one way or another, but I saw we do have `Tensor.is_floating_point` so `Tensor.is_complex` seemed like a reasonable thing to add. There are two different complex types so you are going to want some way to conveniently check if something is complex or not.

**Comment by apaszke** [comment]
I agree we should have a way to check for complex types, but wouldn't it be better to make it a dtype property?

**Comment by anjali411** [comment]
> I agree we should have a way to check for complex types, but wouldn't it be better to make it a dtype property?

yeah I added both

**Pull request closed without merging**